jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java

```diff
@@ -63,28 +60,82 @@
+       |
+       |setWriteListener
+       |
+       v
+      READY ------ write/flush/close ------> PENDING
+       ^ ^                                    |  |
+       | |                                    |  |
+       | +----------isReady==true----------+  |  |
+       |                                   |  |  |
+       |onWriteComplete                    |  |  |onWriteComplete
+       |                                   |  |  |
+       | +----------isReady==false------------+  |
+       | |                                 |     |
+       | v                                 |     v
+      UNREADY                              +---ASYNC
+
+
+          OPEN/BLOCKING---close---------------------------+                      CLOSED/BLOCKING
+         /   |    ^                                        \                         ^  ^
+        /    w    |                                         \                       /   |
+       /     |   owc   +------------------------->--owcL-----\---------------------+    |
+      |      v    |   /                         /             V                         |
+     swl  OPEN/BLOCKED----close--->CLOSE/BLOCKED----owc----->CLOSING/BLOCKED--owcL------+
+       \
+        \
+         \
+          V
+       +->OPEN/READY------close--------------------------+
+      /   ^    |                                          \
+     /   /     w                                           \
+    /   /      |       +------------------------->--owcL----\---------------------------+
+   |   /       v      /                         /            V                          |
+   | irt  OPEN/PENDING----close--->CLOSE/PENDING----owc---->CLOSING/PENDING--owcL----+  |
+   |   \  /    |                        |                    ^     |                 |  |
+  owc   \/    owc                      irf                  /     irf                |  |
+   |    /\     |                        |                  /       |                 |  |
+   |   /  V    V                        |                 /        |                 V  V
+   | irf  OPEN/ASYNC------close---------|----------------+         |             CLOSED/ASYNC
+    \  \                                |                          |                 ^  ^
+     \  \                               |                          |                 |  |
+      \  \                              |                          |                 |  |
+       \  v                             v                          v                 |  |
+        +-OPEN/UNREADY----close--->CLOSE/UNREADY----owc----->CLOSING/UNREADY--owcL---+  |
+                      \                         \                                       |
+                       +------------------------->--owcL--------------------------------+
+
+      swl = setWriteListener
+      w = write
+      owc = onWriteComplete(false,null)
+      owcL = onWriteComplete(true,null)
+      isf = isReady()==false
```

Comment:
This is different to what is on the diagram. This should be irf and irt?

Reply:
yes - fixed